src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -319,7 +319,7 @@ const generateTaskRole = (taskRole, labels, config) => {
         },
         spec: {
           privileged: false,
-          restartPolicy: 'Never',
+          restartPolicy: gangAllocation === 'true' ? 'Never' : 'OnFailure',
```

Comment:
Add comments for this? We should revert this change after we update k8s to 16.2 or above.

Reply:
added